src/main/webapp/app/course/competencies/create/create-competency.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component } from '@angular/core';
+import { onError } from 'app/shared/util/global.utils';
+import { Competency } from 'app/entities/competency.model';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AlertService } from 'app/core/util/alert.service';
+import { CompetencyService } from 'app/course/competencies/competency.service';
+import { finalize } from 'rxjs/operators';
+import { HttpErrorResponse } from '@angular/common/http';
+import { LectureService } from 'app/lecture/lecture.service';
+import { CompetencyFormComponent } from 'app/course/competencies/forms/competency/competency-form.component';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisSharedComponentModule } from 'app/shared/components/shared-component.module';
+import { CreateCourseCompetencyComponent } from 'app/course/competencies/create/create-course-competency.component';
+import { CourseCompetencyFormData } from 'app/course/competencies/forms/course-competency-form.component';
+
+@Component({
+    selector: 'jhi-create-competency',
+    templateUrl: './create-competency.component.html',
+    styles: [],
+    standalone: true,
+    imports: [ArtemisSharedModule, CompetencyFormComponent, ArtemisSharedComponentModule],
+})
+export class CreateCompetencyComponent extends CreateCourseCompetencyComponent {
+    competencyToCreate: Competency = new Competency();
+
+    constructor(
+        activatedRoute: ActivatedRoute,
+        router: Router,
+        alertService: AlertService,
+        lectureService: LectureService,
+        private competencyService: CompetencyService,
+    ) {
+        super(activatedRoute, router, alertService, lectureService);
+    }
+
+    createCompetency(formData: CourseCompetencyFormData) {
+        if (!formData?.title) {
+            return;
+        }
+
+        const { title, description, softDueDate, taxonomy, masteryThreshold, optional, connectedLectureUnits } = formData;
+
+        this.competencyToCreate.title = title;
+        this.competencyToCreate.description = description;
+        this.competencyToCreate.softDueDate = softDueDate;
+        this.competencyToCreate.taxonomy = taxonomy;
+        this.competencyToCreate.masteryThreshold = masteryThreshold;
+        this.competencyToCreate.optional = optional;
+        this.competencyToCreate.lectureUnits = connectedLectureUnits;
+
+        this.isLoading = true;
+
+        this.competencyService
+            .create(this.competencyToCreate!, this.courseId)
+            .pipe(
+                finalize(() => {
+                    this.isLoading = false;
```

Comment:
not sure if the finalize is called after the subscribe. I would move it into the subscribe in the completed block

Reply:
We use this in many places in our code base. I hope it works, otherwise we would need to refactor quite a lot